dev/src/transaction.ts

```diff
@@ -187,8 +191,29 @@ export class Transaction implements firestore.Transaction {
 
     const documentReader = new DocumentReader(this._firestore, documents);
     documentReader.fieldMask = fieldMask || undefined;
-    documentReader.transactionId = this._transactionId;
-    return documentReader.get(this._requestTag);
+
+    if (!this._optimisticLocking) {
+      documentReader.transactionId = this._transactionId;
+    }
+
+    return documentReader.get(this._requestTag).then(docs => {
```

Comment:
Do we have to handle the error case here? If we fail to fetch a document inside a tx, but the error is caught in the transaction callback, the final `commit()` won't have the preconditions set.

I think we need to chain the for loop after a `silencePromise()`.